src/common/reducers/entities.js

```diff
@@ -0,0 +1,66 @@
+import merge from 'lodash/merge';
+
+import * as ActionTypes from '../actions/repository';
+
+// export only for testing
+export function repository(state={
+  __isSelected: false,
+  __isFetching: false,
+  __error: null
+}, action) {
+
+  switch(action.type) {
```

Comment:
it's actually more of a helper function for a reducer, my plan is `entities/index.js` `entities/repository.js`, `entities/snap.js`